src/ui/public/kfetch/default_interceptor.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { merge } from 'lodash';
+// @ts-ignore not really worth typing
+import { metadata } from 'ui/metadata';
+import { Interceptor } from './kfetch';
+
+export const defaultInterceptor: Interceptor = {
```

Comment:
I like the dogfooding and consistency aspect of creating a default interceptor to apply these defaults, but at the same this adds indirection to the code which hurts more than it helps, in my opinion. I'll leave it up to you but personally I think I would find the code easier to follow if we assigned defaults within `kfetch`, as we were doing originally.